nft/src/lib.rs

```diff
@@ -109,6 +115,50 @@ decl_storage! {
 		#[cfg(not(feature = "disable-tokens-by-owner"))]
 		pub TokensByOwner get(fn tokens_by_owner): double_map hasher(twox_64_concat) T::AccountId, hasher(twox_64_concat) (T::ClassId, T::TokenId) => Option<()>;
 	}
+	add_extra_genesis {
+		config(endowed_accounts): BTreeMap<T::AccountId, BTreeMap<(Vec<u8>, T::ClassData), Vec<(Vec<u8>, T::TokenData)>>>;
```

Comment:
this type is super confusing... you can use type alias to improve it. something like

```
type GenesisTokenData<T> = (Vec<u8>, <T as Trait>::TokenData);
type GenesisClassData<T> = (Vec<u8>, <T as Trait>::ClassData);)
```

Also instead of `AccountId => (ClassData => TokenData)`, I will prefer `ClassData => (AccountId, TokenData)`